src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests2.cs

```diff
@@ -0,0 +1,610 @@
+                            }
+                        }
+                        N(SyntaxKind.CloseBraceToken);
+                    }
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void ExtendedPropertySubpattern_06()
+        {
+            UsingExpression(@"e is { a->c: p }");
+
+            N(SyntaxKind.IsPatternExpression);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "e");
+                }
+                N(SyntaxKind.IsKeyword);
+                N(SyntaxKind.RecursivePattern);
+                {
+                    N(SyntaxKind.PropertyPatternClause);
+                    {
+                        N(SyntaxKind.OpenBraceToken);
+                        N(SyntaxKind.Subpattern);
+                        {
+                            N(SyntaxKind.ExpressionColon);
+                            {
+                                N(SyntaxKind.PointerMemberAccessExpression);
+                                {
+                                    N(SyntaxKind.IdentifierName);
+                                    {
+                                        N(SyntaxKind.IdentifierToken, "a");
+                                    }
+                                    N(SyntaxKind.MinusGreaterThanToken);
+                                    N(SyntaxKind.IdentifierName);
+                                    {
+                                        N(SyntaxKind.IdentifierToken, "c");
+                                    }
+                                }
+                                N(SyntaxKind.ColonToken);
+                            }
+                            N(SyntaxKind.ConstantPattern);
+                            {
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "p");
+                                }
+                            }
+                        }
+                        N(SyntaxKind.CloseBraceToken);
+                    }
+                }
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void ExtendedPropertySubpattern_07()
+        {
+            UsingExpression(@"e is { [0]: p }",
+                    // (1,8): error CS1001: Identifier expected
+                    // e is { [0]: p }
+                    Diagnostic(ErrorCode.ERR_IdentifierExpected, "[").WithLocation(1, 8),
+                    // (1,10): error CS1003: Syntax error, ',' expected
+                    // e is { [0]: p }
+                    Diagnostic(ErrorCode.ERR_SyntaxError, "]").WithArguments(",", "]").WithLocation(1, 10),
+                    // (1,13): error CS1003: Syntax error, ',' expected
+                    // e is { [0]: p }
+                    Diagnostic(ErrorCode.ERR_SyntaxError, "p").WithArguments(",", "").WithLocation(1, 13));
```

Comment:
This will be parsed as a length pattern and a missing comma once merged with list patterns.

Reply:
Thanks. The test is useful as its behavior will change once merged. So that will act as a natural reminder.